### ISSUE TITLE
Hangs CPU and takes whole mem

### ISSUE THREAD

**Issue by anna-oake** [Gottox/socket.io-java-client#52]
When I call socket.connect(callback) my java process starts to eat 100% cpu and everything starts lagging.

Implemented methods in callback are empty, so, i'm just connecting and i don't do anything else.

**Comment by anna-oake** [comment]
Using debugger i've discovered that your lib creates many-many threads... And this is a cause of 100% CPU use. But how to fix it?

**Comment by anna-oake** [comment]
Ok, discovered.

It happens when server says that it support Websockets and Xhr, your lib tries to use websockets transport, but server-side WS dont work and lib tries to reconnect many times instead of switching to Xhr. 
We need an option to select transport manually...

**Comment by vinodc** [comment]
This happens for me as well when I restart the server, causing the java process to lose the connection.

**Comment by vinodc** [comment]
I just tested by cloning https://github.com/TooTallNate/Java-WebSocket/ and building the jar and copying it into libs and then building socket.io-java-client and it seems to have resolved the issue for me. I guess all we need to do is upgrade the version of Java-WebSocket socket.io-java-client uses? Your issue is slightly different from mine though, so I'm not sure if this would work for you.

**Comment by anna-oake** [comment]
My problem was because server doesnt support websockets. 
But it doesn't matter, the problem is that Java-WebSocket lib can't handle errors correctly.
You updated to newer version of Websocket and it works, so it means that they just made better error handling.

Author of this lib should just update the dependence

**Comment by volpino** [comment]
@vinodc I tried to update Java-WebSocket but nothing changes.
I have the issue if something fails in the process, for example:
1) I connect
2) the server gives me an error
3) I disconnect
4) I reconnect again

I have the CPU at 60% if this happens, otherwise if everything goes fine and I can access the server without any error the CPU is maximum at 4%

I'm using Android 4.2

**Comment by volpino** [comment]
I noticed that this issue is due to the call of `socket.disconnect()`. If that method is not called everything works fine and the CPU usage is very low

**Comment by emmgfx** [comment]
Yes, I think it's a problem with the disconnect method. At least in part.

**Comment by groff** [comment]
Someone solved this problem?

**Comment by DHTMLGoodies** [comment]
I'm experiencing the same problem with

``` java
socket.disconnect();
```

Once disconnect is called, the CPU jumps from 1-2% to 25 - 80 %

It would be nice to get a fix for this problem/bug :)

**Comment by dwalintukan** [comment]
Anyone ever find a fix to the socket.disconnect() bug? I am getting the same issue. CPU usage is fine until I call disconnect().

**Comment by fiskbil** [comment]
This is actually a bug in the Java-WebSocket library that this lib uses. You can fix it by updating Java-WebSocket to 1.3.0:
1. git clone git@github.com:TooTallNate/Java-WebSocket.git
2. cd Java-WebSocket
3. git checkout tags/Java-WebSocket-1.3.0
4. ant jar
5. cp dist/java_websocket.jar _[path to socket.io-java-client]_/libs/WebSocket.jar
6. rebuild the socket.io-java-client lib as usual

**Comment by ghost** [comment]
Thx It's work !!
But I have a new issue
My App will crash when network switching (3G to Wifi) causing this.conn in WebSocketClinet.class is null.(Line 223)
So I found new java-websocket.jar from http://java-websocket.org/ , It's work for network switching but it consumes CPU 
May you give me a version that has clear both issues.

Thanks
Sirisub,

**Comment by anna-oake** [comment]
You should ask @Davidiusdadi about that, @sirisub
